pkg/grpcserver/server.go

```diff
@@ -0,0 +1,154 @@
+package grpcserver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"google.golang.org/grpc"
+	_ "google.golang.org/grpc/encoding/gzip"
+
+	"github.com/justtrackio/gosoline/pkg/cfg"
+	"github.com/justtrackio/gosoline/pkg/kernel"
+	"github.com/justtrackio/gosoline/pkg/log"
+
+	protobuf "github.com/justtrackio/gosoline/pkg/grpcserver/proto/health/v1"
+)
+
+const (
+	grpcServerConfigKey = "grpc_server"
+	grpcServiceChannel  = "grpc_service"
+)
+
+// Server a basic grpc.Server wrapper that allows also can have a basic Health Check functionality.
+type Server struct {
+	kernel.EssentialModule
+	kernel.ServiceStage
+
+	logger       log.Logger
+	listener     net.Listener
+	serverCtx    context.Context
+	cancelFunc   context.CancelFunc
+	server       *grpc.Server
+	healthServer *healthServer
+}
+
+type Middleware grpc.UnaryServerInterceptor
+type MiddlewareFactory func(logger log.Logger) Middleware
+
+// New returns a kernel.ModuleFactory for the Server kernel.Module.
+func New(name string, definer ServiceDefiner, middlewares ...MiddlewareFactory) kernel.ModuleFactory {
+	return func(ctx context.Context, config cfg.Config, logger log.Logger) (kernel.Module, error) {
+		var (
+			err         error
+			definitions *Definitions
+		)
+		settings := &Settings{}
+		config.UnmarshalKey(fmt.Sprintf("%s.%s", grpcServerConfigKey, name), settings)
+
+		logger = logger.WithChannel(grpcServiceChannel)
```

Comment:
Do you want to have the server name in the channel?  
Or add the server name as a field?  
To be able to distinguish between different servers?

Reply:
I think it would be better to have it as a field, will add it